back/engines/commercial/bulk_import_ideas/app/services/bulk_import_ideas/import_global_ideas_service.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+module BulkImportIdeas
+  class ImportGlobalIdeasService < ImportIdeasService
+    def generate_example_xlsx
+      XlsxService.new.hash_array_to_xlsx [
+        {
+          'ID' => '1',
+          'Title_nl-BE' => 'Mijn idee titel',
+          'Title_fr-BE' => 'Mon idée titre',
+          'Body_nl-BE' => 'Mijn idee inhoud',
+          'Body_fr-BE' => 'Mon idée contenu',
+          'Email' => 'moderator@citizenlab.co',
+          'Full name' => 'Paul Moderator',
+          'Permission' => 'X',
+          'Project' => 'Project 1',
+          'Phase' => 1,
+          'Image URL' => 'https://cl2-seed-and-template-assets.s3.eu-central-1.amazonaws.com/images/people_in_meeting_graphic.png',
+          'Date (dd-mm-yyyy)' => '18-07-2022',
+          'Topics' => 'Mobility; Health and welfare',
+          'Latitude' => 50.5035,
+          'Longitude' => 6.0944,
+          'Location Description' => 'Panorama sur les Hautes Fagnes / Hohes Venn'
+        }
+      ]
+    end
+
+    def parse_idea_rows(file, _file_type)
+      xlsx_ideas = parse_xlsx_ideas file
+      ideas_to_idea_rows xlsx_ideas
+    end
+
+    def ideas_to_idea_rows(ideas_array)
+      idea_rows = ideas_array.map do |xlsx_row|
+        next if idea_blank? xlsx_row
+
+        idea_row = {}
+
+        title_multiloc = {}
+        body_multiloc  = {}
+        xlsx_row.each do |key, value|
+          next unless key.include? '_'
+
+          field, locale = key.split '_'
+          raise Error.new 'bulk_import_ideas_locale_not_valid', value: locale if AppConfiguration.instance.settings('core', 'locales').exclude?(locale)
```

Comment:
Calling `AppConfiguration.instance` usually triggers an SQL query or hits the cache, it's often possible to make a significant performance gain when instantiating the app configuration outside of the loop.